Dev blockchain + (blockchain v2): wire frontend to v2 contracts

zohersahli opened the pull request.

zohersahli committed:
Frontend + Blockchain dev

zohersahli committed:
feat(blockchain): wire frontend to v2, add v2 foundry.toml, adjust CreatePoll/Proposal, update pollsBlockchain + env notes

zohersahli committed:
feat: membership logging + fix becomeMember flow

zohersahli committed:
Added Blockchain v2 documentation, adapters, JS modules, and updated .gitignore

zohersahli committed:
Add polls adapter for v2 and fix poll creation

- Add pollsAdapter.ts for unified v1/v2/REST polls support
- Fix createPoll to handle receipt.status as number or BigInt
- Update CreatePoll.svelte to use pollsAdapter
- Update blockchain test page to use unified adapters
- Add comprehensive logging for poll creation debugging

zohersahli committed:
Merge upstream/dev into dev-blockchain

Kattenelvis commented on src/lib/Blockchain_v1_Ethereum/javascript/pollsBlockchain.ts:
```diff
@@ -1,15 +1,20 @@
 interface Window {
-	ethereum?: import('ethers').providers.ExternalProvider;
+	//ethereum?: import('ethers').providers.ExternalProvider;
+	ethereum?: any;
```
Try to avoid "any"

Kattenelvis commented on src/lib/Blockchain_v2_CrossChain/foundry.toml:
```diff
@@ -1,6 +1,28 @@
+optimizer = true
+optimizer_runs = 20000
+via_ir = true
+evm_version = "paris"
```
Check out if testnets, aswell as the ethereum net itself, can handle contracts deployed with earlier versions on newer chains.

zohersahli committed:
Improve wallet connection reliability in v2 client

- Add EIP-1193 window.ethereum typing support (global + local event provider typing).

- Use eth_accounts first, then eth_requestAccounts only when needed.

- Handle user-rejected wallet connection (error code 4001) with explicit message.

- Validate active chainId against PUBLIC_V2_CHAIN_ID before creating signer.

- Reset cached signer/provider on accountsChanged and chainChanged to prevent stale wallet state.

zohersahli committed:
Adjust gitignore: ignore Foundry lib, track frontend src/lib

zohersahli committed:
Update gitignore: track src/lib, ignore Foundry artifacts

zohersahli committed:
Add v2 deploy scripts and polls adapter

zohersahli committed:
Ignore v1 Foundry config/scripts for now

zohersahli committed:
Refactor v2 delegations (read/write separation)

- Reuse shared wallet/contract helpers from client.ts

- Use RPC provider for read-only calls (no MetaMask popups)

- Cache read-only Contract instance to avoid re-creating provider

- Keep write calls via signer-backed Polls contract
